signer/src/transaction_signer.rs

```diff
@@ -0,0 +1,69 @@
+//! # Transaction signer
+//!
+//! This module contains the transaction signer, which is
+//! responsible for preparing and participating in signing rounds.
+//!
+//! For more details, see the [`EventLoop`] documentation.
+
+#[cfg_attr(doc, aquamarine::aquamarine)]
+/// # Transaction signer event loop
+///
+/// This struct contains the implemetnation of the transaction signer logic.
+/// The event loop subscribes to storage update notifications from the block observer,
+/// and listens to signer messages from the signer entwork.
+///
+/// ## On storage update
+/// When the signer receives a storage update notification, it must go over each of the pending
+/// requests and decide wheter to accept or reject it. The decision is then persisted
+/// and broadcast to the other signers. The following flowchart illustrates the flow.
+///
+/// ```mermaid
+/// flowchart TD
+///     SU{Storage update notification} --> FPR(Fetch pending requests)
+///     FPR --> NR(Next request)
+///     NR --> |deposit/withdraw| DAR(Decide to accept/reject)
+///     NR ----> |none| DONE{Done}
+///     DAR --> PD(Persist decision)
+///     PD --> BD(Broadcast decision)
+///     BD --> NR
+/// ```
+///
+/// ## On signer message
+/// When the signer receives a signer message, it needs to do a few different things
+/// depending on the type of the message.
+///
+/// - **Signer decision**: When receiving a signer decision, the transaction signer
+/// only need to persist the decision to the signer storage.
+/// - **Stacks sign request**: When receiving a request to sign a stacks transaction,
+/// the signer must verify that it has decided to sign the transaction, and if positive,
+/// send a transaction signature back over the network.
+/// - **Bitcoin sign request**: When receiving a request to sign a bitcoin transaction,
+/// the signer must verify that it has decided to accept all request that the
+/// transaction fulfills. Once verified, the transaction signer creates a dedicated
```

Comment:
Is the transaction signer always the coordinator?

Reply:
I view the coordinator is a separate entity with it's own event loop (as hinted in #44), but I am open to other designs. I created #222 to write down a design for the coordinator event loop.

Reply:
We spoke offline about this and the transaction signer process/code path is supposed to be different from the coordinator. That design makes sense to me. Resolved!